src/models/task/model.py

```diff
@@ -0,0 +1,147 @@
+class TaskOrientedChatbot:
+    def __init__(self):
+        self.done = False
+        self.collected_data = self._init_data()
+        self.stage = 0
+        "possible stages:\n"
+        "        0. ask to confirm task start\n"
+        "        1. confirm start\n"
+        "        2. ask for information\n"
+        "           provide answer\n"
+        "        3. confirm end\n"
+        "           ask for rechecking\n"
+        "        4. confirm rechecking\n"
+        "        5. recheck information\n"
+        self.current_field = None
+
+
+    def _is_completed(self):
+        return self.done
+
+    @staticmethod
+    def _init_data():
+        return dict(
+            group=dict(
+                data_type=int,
+                value=None,
+                question="O którą grupę pytasz?",
+                priority=1,
+                check=False
+            ),
+            tutor=dict(
+                data_type=int,
+                value=None,
+                question="Kto prowadzi zajęcia?",
+                priority=1,
+                check=False
+            ),
+            course=dict(
+                data_type=int,
+                value=None,
+                question="O jaki przedmiot pytasz?",
+                priority=1,
+                check=False
+            ),
+            room=dict(
+                data_type=int,
+                value=None,
+                question="W jakiej sali odbywają się zajęcia?",
+                priority=1,
+                check=False
+            ),
+            start_hour=dict(
+                data_type=int,
+                value=None,
+                question="O której rozpoczynają się zającia?",
+                priority=1,
+                check=False
+            )
+        )
+
+    def _detect_confirmation(self, prompt):
+        if 'tak' in prompt:
+            return True
+        elif 'nie' in prompt:
+            return False
+        return None
+
+    def _ask_for_data(self):
+        for k, v in self.collected_data.items():
+            if v["value"] is None:
+                self.current_field = k
+                return v['question']
+
+    def _retrive_info(self, prompt):
+        0  # TODO
+
+
+    def _check_answer(self, recheck=False):
+        return None
+
+    def _check_data(self):
+        for k, v in self.collected_data.items():
+            if not v["check"]:
+                self.current_field = k
+                return f"Czy {k} to {v['value']}"  # TODO: provide polish names
+        return None
+
+    def _reset(self):
+        self.done = False
+        self.collected_data = self._init_data()
+        return "To o czym my tu rozmawialiśmy?"
+
+    def interact(self, prompt):
```

Comment:
This method is a bit difficult to understand at the first glance - can you add a comment describing what happens/what is the flow here?

Reply:
Now I see the comment in `__init__` - I think it would be more readable if we move it here, WDYT?